indexd/src/quic/upload.rs

```diff
@@ -346,7 +347,11 @@ impl Uploader {
         }
         read_slab_res.await??;
 
-        let object = Object::new(slabs, meta);
+        let nonce_meta = nonce_prefix
+            .into_iter()
+            .chain(meta.unwrap_or_default())
+            .collect();
+        let object = Object::new(slabs, Some(nonce_meta));
```

Comment:
since the metadata will always contain the nonce prefix, we can get rid of the `Option` in the object's constructor

Reply:
Should we update the object's constructor to take a nonce_prefix and metadata instead? So merging the two happens in the constructor instead. We should probably also extend `validate_object` to ensure its metadata is at least 16 bytes.